From c4eaef426c18c6f50a7d4a9e2423476314820f31 Mon Sep 17 00:00:00 2001
From: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
Date: Thu, 16 Sep 2010 17:03:53 -0700
Subject: [PATCH 14/26] xen/apic: fix pirq_eoi_gmfn resume

Only BUG if the hypercall *fails*...

Signed-off-by: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
(cherry picked from commit f0d4a0552f03b52027fb2c7958a1cbbe210cf418)

Signed-off-by: Ian Campbell <ijc@hellion.org.uk>
---
 drivers/xen/events.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/xen/events.c b/drivers/xen/events.c
index 948c437..5191db7 100644
--- a/drivers/xen/events.c
+++ b/drivers/xen/events.c
@@ -1365,7 +1365,7 @@ void xen_irq_resume(void)
 		struct physdev_pirq_eoi_gmfn eoi_gmfn;
 		
 		eoi_gmfn.gmfn = virt_to_mfn(pirq_needs_eoi_bits);
-		if (HYPERVISOR_physdev_op(PHYSDEVOP_pirq_eoi_gmfn, &eoi_gmfn) == 0) {
+		if (HYPERVISOR_physdev_op(PHYSDEVOP_pirq_eoi_gmfn, &eoi_gmfn) != 0) {
 			/* Could recover by reverting to old method...? */
 			BUG();
 		}
-- 
1.7.5.4

